qa/L0_cuda_shared_memory/cuda_shared_memory_test.py

```diff
@@ -397,171 +409,222 @@ def test_infer_byte_size_out_of_bound(self):
         self.assertIn(
             "Invalid offset + byte size for shared memory region", error_msg[0]
         )
-        self._cleanup_server(shm_handles)
+        self._cleanup_shm_handles()
 
 
 class TestCudaSharedMemoryUnregister(CudaSharedMemoryTestBase):
-    def _test_unregister_shm_fail(self):
+    def _test_unregister_shm_request_pass(self, shm_names):
+        self._test_shm_found(shm_names)
+
+        # Unregister all should not result in an error.
+        # If shared memory regions are in use, they will be marked and unregistered after the inference is completed.
+        with httpclient.InferenceServerClient(
+            "localhost:8000", verbose=True
+        ) as second_client:
+            second_client.unregister_cuda_shared_memory()
+
+        # Number of shared memory regions should be the same as the inference is not completed yet
+        self._test_shm_found(shm_names)
```

Comment:
So the region will still be visible between unregister and completion of the inference, what would happen if another inference request is sent during that time, trying to reuse the unregistered region? Can you add a test for this case to demonstrate the behavior?

Reply:
Yes, the shm will remain accessible between unregister and the completion of the inference. When we run another inference request that reuses the unregistered region, the shm region will be extended and unregistered after the second request, as we check the [reference count](https://github.com/triton-inference-server/server/pull/7743/files#diff-b2df344fcbed5fc3756d51cd3ea194ac9edecbcc7a6f40022642520c92cf8f78L688) before unregister. Added a test case to verify this.